### ISSUE TITLE
Helm 3-alpha1: pushing chart without tag specified results in untagged chart instead of latest

### ISSUE THREAD

**Issue by nphmuller** [helm/helm#5752]
Output of `helm version`:
`version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}`

Output of `kubectl version`:
`Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"windows/amd64"}`

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Azure Container Registry (but should be reproducable on any OCI compatible registry)

Repro:
- `helm create test`
- `helm chart save test myreg.azurecr.io/test` (replace registry url)
- `helm chart push myreg.azurecr.io/test`

Output `helm chart push` command:
```
Using default tag: latest
The push refers to repository [myreg.azurecr.io/test]
Name: test
Version: 0.1.0
Meta: sha256:xxx
Content: sha256:xxx
latest: pushed to remote (2 layers, 2.3 KiB total)
```

Expected: 
The output indicates the the default tag (`latest`) is used, so it's expected that the remote registry now contains a `test` repository with a manifest tagged `latest`.

Actual:
The remote registry now contains a `test` repository with a single manifest, but its untagged (or at least not tagged latest).

**Comment by bacongobbler** [comment]
cc @jdolitsky

**Comment by jdolitsky** [comment]
@nphmuller thanks for opening, taking a look today

**Comment by muehlburger** [comment]
@nphmuller and @jdolitsky would be happy to get feedback from you.

**Comment by nphmuller** [comment]
@muehlburger I'll try it out next week!

**Comment by jdolitsky** [comment]
@muehlburger @nphmuller the maintainers are currently considering whether we even want to allow for tags such as "latest", and instead enforce semantic versioning based on `version` field in Chart.yaml

**Comment by jdolitsky** [comment]
It would probably be preferred default behavior that if a tag is missing, we source it from Chart.yaml

**Comment by muehlburger** [comment]
@jdolitsky thank you for your feedback. So there are the following requirements:

1. Do not allow "latest" tag but enforce semantic versioning based on `version` field in Chart.yaml
2. If tag is missing, source it from `version` field in Chart.yaml

What if `version` field in Chart.yaml is missing? I guess this is an invalid chart?

**Comment by nphmuller** [comment]
I'd be fine with using the `version` field from `Chart.yaml`. Seems like a fine solution!

@muehlburger According to the spec in https://helm.sh/docs/developing_charts/ (not sure how much it changes for v3), version would be required, so my guess would be the chart is invalid.

**Comment by jdolitsky** [comment]
@muehlburger @nphmuller this all sounds correct. The version/tag should be sourced from Chart.yaml and should be a valid semantic version (there is already code that validates this field).

An option `--version` override flag can be discussed, such as the one that exists on `helm package` (Helm 2):
```
--version string       set the version on the chart to this semver version
```

However, this can come later on. Feel free to ping me on CNCF slack to discuss further: https://slack.cncf.io/

**Comment by zioproto** [comment]
I have the same problem here with Google Container Registry (GCR).
I dont care having `latest`, but can anyone clarify how to run `helm chart push` and have a pushed chart in the registry with a tag ?
Because without any tag I am not able to `helm chart pull`.
Thanks

**Comment by hickeyma** [comment]
@nphmuller @jdolitsky @muehlburger Would  `helm chart pull` also be covered by this?  Refer to #5906

**Comment by bacongobbler** [comment]
ping, anyone planning to work on this? We're planning to cut the first beta by mid-August so it'd be great if we could take a closer look at these design issues.

**Comment by jdolitsky** [comment]
Hi, I can take a look in the coming days

**Comment by muehlburger** [comment]
@jdolitsky I can also take a look in the coming days.

**Comment by jdolitsky** [comment]
Hi all, this should essentially be fixed with #6059 . Please reach out directly (slack etc.) if needed

**Comment by jdolitsky** [comment]
@muehlburger if you're interested in development on the registry side, let's chat further

**Comment by muehlburger** [comment]
@jdolitsky sure, what would be the best part to work on? I'd also be happy to contribute on improving tests to get to know the codebase better.

**Comment by bacongobbler** [comment]
fixed via #6070

**Comment by jdolitsky** [comment]
Thanks @bacongobbler !

@muehlburger - I've reached out via email to discuss further
